src/components/Autocomplete/components/ComboBox/ComboBox.tsx

```diff
@@ -53,467 +42,337 @@ export interface ComboBoxProps {
+      }
+    },
+    [
+      allowMultiple,
+      forcePopoverActiveFalse,
+      onSelect,
+      resetVisuallySelectedOptions,
+    ],
+  );
 
-    const {selectedOption} = this.state;
-    if (this.state.popoverActive && selectedOption) {
-      if (isOption(selectedOption)) {
-        event.preventDefault();
-        this.handleSelection(selectedOption.value);
+  const handleSelection = useCallback(
+    (newSelected: string) => {
+      let newlySelectedOptions = selected;
+      if (selected.includes(newSelected)) {
+        newlySelectedOptions.splice(
+          newlySelectedOptions.indexOf(newSelected),
+          1,
+        );
+      } else if (allowMultiple) {
+        newlySelectedOptions.push(newSelected);
       } else {
-        selectedOption.onAction && selectedOption.onAction();
+        newlySelectedOptions = [newSelected];
       }
-    }
 
-    this.handlePopoverOpen;
-  };
-
-  private handleFocus = () => {
-    this.setState({popoverActive: true, popoverWasActive: true});
-  };
+      selectOptions(newlySelectedOptions);
+    },
+    [allowMultiple, selectOptions, selected],
+  );
 
-  private handleBlur = () => {
-    this.setState({popoverActive: false, popoverWasActive: false}, () => {
-      this.resetVisuallySelectedOptions();
-    });
-  };
+  const handleDownArrow = useCallback(() => {
+    selectNextOption();
+  }, [selectNextOption]);
 
-  private handleClick = () => {
-    !this.state.popoverActive && this.setState({popoverActive: true});
-  };
+  const handleUpArrow = useCallback(() => {
+    selectPreviousOption();
+  }, [selectPreviousOption]);
 
-  private handleSelection = (newSelected: string) => {
-    const {selected, allowMultiple} = this.props;
-    let newlySelectedOptions = selected;
-    if (selected.includes(newSelected)) {
-      newlySelectedOptions.splice(newlySelectedOptions.indexOf(newSelected), 1);
-    } else if (allowMultiple) {
-      newlySelectedOptions.push(newSelected);
-    } else {
-      newlySelectedOptions = [newSelected];
-    }
+  const handleEnter = useCallback(
+    (event: KeyboardEvent) => {
+      if (event.keyCode !== Key.Enter) {
+        return;
+      }
 
-    this.selectOptions(newlySelectedOptions);
-  };
+      if (popoverActive && selectedIndex > -1) {
+        const selectedOption = navigableOptions[selectedIndex];
+        if (isOption(selectedOption)) {
+          event.preventDefault();
+          handleSelection(selectedOption.value);
+        } else {
+          selectedOption.onAction && selectedOption.onAction();
+        }
+      }
+    },
+    [handleSelection, navigableOptions, popoverActive, selectedIndex],
+  );
 
-  private selectOptions = (selected: string[]) => {
-    const {onSelect, allowMultiple} = this.props;
-    selected && onSelect(selected);
-    if (!allowMultiple) {
-      this.resetVisuallySelectedOptions();
-      this.setState({popoverActive: false, popoverWasActive: false});
-    }
-  };
+  const handleFocus = useCallback(() => {
```

Comment:
As above, forcePopoverActiveTrue is already afunction wrapped in a callback, replace usages of `handleFocus` with `forcePopoverActiveTrue` and this can be removed